# Relatório de Análise de Negócios com Uso de Big Data

<ul>
    <li>nome: Pedro Gomes Vieira Mazone matrícula: 1220105326</li>
    <li>nome: Gustavo da Silva Medina Teixeira matrícula: 1220202550</li>
    <li>nome: Germano S Junior matrícula: 1220202308</li>
    <li>dataset : https://www.kaggle.com/datasets/computingvictor/top1000youtubers </li>
</ul>

### Já vimos diversos criadores de conteúdo dizerem que o youtube paga um valor mais alto em adSense, para criadores americanos, do que para criadores do resto do mundo. Desta forma fizemos um levantamento de um dataset disponibilizado pelo Youtube com os principais canais da plataforma

## Dataframe

In [87]:
# Importando as bibliotecas
import pandas as pd
import plotly.express as px

In [88]:
# Leitura do arquivo xlsx
df = pd.read_csv('youtubers_df.csv')

# Exibir o DataFrame para visualização
display(df)

Rank              Username           Categories   Suscribers  \
0       1               tseries       Música y baile  249500000.0   
1       2               MrBeast   Videojuegos, Humor  183500000.0   
2       3             CoComelon            Educación  165500000.0   
3       4              SETIndia                  NaN  162600000.0   
4       5         KidsDianaShow  Animación, Juguetes  113500000.0   
..    ...                   ...                  ...          ...   
995   996          hamzymukbang                  NaN   11700000.0   
996   997            Adaahqueen                  NaN   11700000.0   
997   998  LittleAngelIndonesia       Música y baile   11700000.0   
998   999          PenMultiplex                  NaN   11700000.0   
999  1000         OneindiaHindi  Noticias y Política   11700000.0   

            Country       Visits      Likes  Comments  \
0             India      86200.0     2700.0      78.0   
1    Estados Unidos  117400000.0  5300000.0   18500.0   
2           Unknown    7000000.0    24700.0       0.0   
3             India      15600.0      166.0       9.0   
4           Unknown    3900000.0    12400.0       0.0   
..              ...          ...        ...       ...   
995  Estados Unidos     397400.0    14000.0     124.0   
996           India    1100000.0    92500.0     164.0   
997         Unknown     211400.0      745.0       0.0   
998           India      14000.0       81.0       1.0   
999           India       2200.0       31.0       1.0   

                                                 Links  
0    http://youtube.com/channel/UCq-Fj5jknLsUf-MWSy...  
1    http://youtube.com/channel/UCX6OQ3DkcsbYNE6H8u...  
2    http://youtube.com/channel/UCbCmjCuTUZos6Inko4...  
3    http://youtube.com/channel/UCpEhnqL0y41EpW2TvW...  
4    http://youtube.com/channel/UCk8GzjMOrta8yxDcKf...  
..                                                 ...  
995  http://youtube.com/channel/UCPKNKldggioffXPkSm...  
996  http://youtube.com/channel/UCk3fFpqI5kDMf__mUP...  
997  http://youtube.com/channel/UCdrHrQf0o0TO8YDntX...  
998  http://youtube.com/channel/UCObyBrdrtQ20BU9PxH...  
999  http://youtube.com/channel/UCOjgc1p2hJ4GZi6pQQ...  

[1000 rows x 9 columns]

<p>A TOP 1000 apresenta os youtubers mais famosos e seus números.</p>

## Limitamos Os dados para as seguintes categorias:
<ul>
    <li>Username: Nome dos canais</li>
    <li>Categories: Categoria em que este criador atua</li>
    <li>Country: País de origem dos canais</li>
    <li>visits: Visualizações do canal</li>
    <li>Likes: As curtidas dos canais</li>
    <li>Subscribers_Millions: Variável criada para facilitar a visualização dos inscritos</li>
</ul>

In [89]:
# Remove as colunas Links, Comments, e Rank
df = df.drop('Links',axis = 1)
df = df.drop('Comments',axis = 1)
df = df.drop('Rank', axis=1)

# Calcula a coluna 'Subscribers_millions' dividindo a coluna 'Suscribers' por 1.000.000
df['Subscribers_millions'] = df['Suscribers'] / 1000000
# Remove a Coluna Subscribes
df = df.drop('Suscribers',axis = 1)

# Exibir o DataFrame após as operações de remoção e cálculo
display(df)

Username           Categories         Country       Visits  \
0                 tseries       Música y baile           India      86200.0   
1                 MrBeast   Videojuegos, Humor  Estados Unidos  117400000.0   
2               CoComelon            Educación         Unknown    7000000.0   
3                SETIndia                  NaN           India      15600.0   
4           KidsDianaShow  Animación, Juguetes         Unknown    3900000.0   
..                    ...                  ...             ...          ...   
995          hamzymukbang                  NaN  Estados Unidos     397400.0   
996            Adaahqueen                  NaN           India    1100000.0   
997  LittleAngelIndonesia       Música y baile         Unknown     211400.0   
998          PenMultiplex                  NaN           India      14000.0   
999         OneindiaHindi  Noticias y Política           India       2200.0   

         Likes  Subscribers_millions  
0       2700.0                 249.5  
1    5300000.0                 183.5  
2      24700.0                 165.5  
3        166.0                 162.6  
4      12400.0                 113.5  
..         ...                   ...  
995    14000.0                  11.7  
996    92500.0                  11.7  
997      745.0                  11.7  
998       81.0                  11.7  
999       31.0                  11.7  

[1000 rows x 6 columns]

In [90]:
# Agrupa o Dataframe por 'Country' e soma os valores da coluna 'Subscribers_million'
inscritos_por_pais = df.groupby('Country')['Subscribers_millions'].sum().reset_index()

# Ordena o novo DataFrame em ordem decrescente com base na coluna 'Subscribers_millions'
inscritos_por_pais = inscritos_por_pais.sort_values(by='Subscribers_millions', ascending=False)

# Reseta o índice do DataFrame 'inscritos_por_pais' para criar um novo índice numérico
inscritos_por_pais = inscritos_por_pais.reset_index(drop=True)

## Grafico em Barra
Com base no DataFrame criamos o gráfico de países por milhões de subscritos.

In [91]:
# plot o grafico em barra usando o Df 'inscritos_por_pais'
grafico = px.bar(inscritos_por_pais, x=inscritos_por_pais.Country, y=inscritos_por_pais.Subscribers_millions,
                title='Numero de subscritos por país')

# Exibe o grafico
grafico.show()

Os Estados Unidos lideram o ranking de inscritos tendo 442.9 pontos a mais que a Índia. Essa tabela explicita a diferença de inscrições dos maiores “consumidores” da plataforma o top 25 tem os EUA em primeiro lugar e a Somália em 25º e a diferença entre eles sendo de 6254.9 pontos a Somália tendo apenas 11.7 milhões de inscritos. Além disso, o  Brasil que se encontra em 4º no ranking, está mais próximo da última colocada Somália, com uma diferença de 1235.3, do que se comparado ao 1º Colocado EUA, com uma diferença de aproximadamente 5200 pontos.

### Criação de uma nova coluna

In [92]:
# Cria uma nova coluna no df que chama 'likes/visit', que calcula a divisa das colunas 'likes' por 'visits'
df['likes/visit'] = df['Likes'] / df['Visits']

# exibe o DF apos a adição da nova coluna 
display(df)

Username           Categories         Country       Visits  \
0                 tseries       Música y baile           India      86200.0   
1                 MrBeast   Videojuegos, Humor  Estados Unidos  117400000.0   
2               CoComelon            Educación         Unknown    7000000.0   
3                SETIndia                  NaN           India      15600.0   
4           KidsDianaShow  Animación, Juguetes         Unknown    3900000.0   
..                    ...                  ...             ...          ...   
995          hamzymukbang                  NaN  Estados Unidos     397400.0   
996            Adaahqueen                  NaN           India    1100000.0   
997  LittleAngelIndonesia       Música y baile         Unknown     211400.0   
998          PenMultiplex                  NaN           India      14000.0   
999         OneindiaHindi  Noticias y Política           India       2200.0   

         Likes  Subscribers_millions  likes/visit  
0       2700.0                 249.5     0.031323  
1    5300000.0                 183.5     0.045145  
2      24700.0                 165.5     0.003529  
3        166.0                 162.6     0.010641  
4      12400.0                 113.5     0.003179  
..         ...                   ...          ...  
995    14000.0                  11.7     0.035229  
996    92500.0                  11.7     0.084091  
997      745.0                  11.7     0.003524  
998       81.0                  11.7     0.005786  
999       31.0                  11.7     0.014091  

[1000 rows x 7 columns]

Um fator importante para os canais é sua “avaliação”, que no Youtube é conhecida por like/curtida. O like ajuda o algoritmo a entender que o usuário ao assistir o vídeo, gostou do conteúdo, sendo assim, ele deve indicá-lo para mais pessoas a fim de tentar prender a audiência no site. E como se pode ver, os maiores canais possuem uma taxa de curtida por visita muito baixa, ponto que poderia ser revisto, pois dentre o top 5, o canal com maior media é o de mrBeast com 4%.

### Agrupamento e Criação da tabela Country likes/visit

Nessa tabela fizemos um agrupamento do dataframe por Países e somando os valores da coluna like/visit.


In [93]:
# Agrupa o Dataframe por 'Country' e soma os valores da coluna 'likes/visit'
pais_por_like_e_visit = df.groupby('Country')['likes/visit'].sum().reset_index()

# Ordena o novo DataFrame em ordem decrescente com base na coluna 'likes/visit'
pais_por_like_e_visit = pais_por_like_e_visit.sort_values(by='likes/visit', ascending=False)

# Reseta o índice do DataFrame 'pais_por_like_e_visit' para criar um novo índice numérico
pais_por_like_e_visit = pais_por_like_e_visit.reset_index(drop=True)

# Exibir o DataFrame 'pais_por_like_e_visit' após as operações de agrupamento, ordenação e redefinição de índice
display(pais_por_like_e_visit)

Country  likes/visit
0   Estados Unidos    12.024883
1            India     8.767196
2           Brasil     4.369728
3           México     3.290259
4          Unknown     1.658101
5        Indonesia     1.154530
6            Rusia     0.928349
7         Colombia     0.911219
8        Filipinas     0.438965
9        Tailandia     0.420622
10       Argentina     0.357938
11            Iraq     0.309030
12          España     0.263941
13        Pakistán     0.182788
14          Egipto     0.132566
15         Turquía     0.132515
16         Ecuador     0.125364
17         Argelia     0.115132
18         Francia     0.100846
19  Arabia Saudita     0.099195
20     Reino Unido     0.067902
21     El Salvador     0.054853
22      Bangladesh     0.053645
23           Japón     0.052708
24         Somalia     0.051842
25            Perú     0.049823
26       Marruecos     0.033000
27        Singapur     0.024205
28        Jordania     0.007100

### Grafico em Barra

In [94]:
# plota um grafico em forma de barra usando o DF 'df'
grafico_like_e_visit = px.bar(pais_por_like_e_visit, x='Country', y='likes/visit',
                             title='Numero da soma de likes/visit por país')
# Exibe o grafico
grafico_like_e_visit.show()

este Gráfico mostra a média de like e visitante de cada país  quantas se tratando de quantas pessoas em média entra no canal visualiza e não deixa o like sendo os Estados unidos liderando neste gráfico.

### Cálculo de indicadores matemáticos ( média)

Nessa parte do projeto foi feito o cálculo de indicadores matemáticos de tendência central como: média de todas as colunas numéricas do Dataframe.

In [95]:
# Calcula a media das colunas numericas do df e arredonda para duas casas decimais
df.mean().round(2)

C:\Users\BISCOITO DOCE\AppData\Local\Temp\ipykernel_736\1431295192.py:2: FutureWarning:

The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.



Visits                  1209446.32
Likes                     53632.59
Subscribers_millions         21.89
likes/visit                   0.04
dtype: float64

### Tabela e Gráfico Relacionada ao Top 20 youtubers
Nessa partes nós fizemos o agrupamento do Dataframe para pegar a coluna 'username' e somar as colunas 'likes' e depois ordenar os 20 com maiores likes.

In [96]:
# Agrupa o df pela coluna de 'Username' e soma as colunas de 'Likes'
usuarios_por_likes = df.groupby('Username')['Likes'].sum().reset_index()
# ordena o DF em ordem decrescente e seleciona os 20 primeiros Username com mais Likes
usuarios_por_likes = usuarios_por_likes.sort_values(by='Likes', ascending=False).head(20)

# Redefine o indice
usuarios_por_likes = usuarios_por_likes.reset_index(drop=True)

#Exibe o DF 'usuario_por_likes' apos realizar as operações de agrupamento, ordenação e redefinição de indice
display(usuarios_por_likes)

Username      Likes
0                    MrBeast  5300000.0
1                   MrBeast2  5000000.0
2                  DaFuqBoom  1700000.0
3            triggeredinsaan  1400000.0
4                    Mythpat  1200000.0
5          BeastPhilanthropy   952100.0
6                   _vector_   712400.0
7               alfredolarin   707600.0
8             MRINDIANHACKER   617400.0
9                 alanbecker   582600.0
10      ShortsBreak_Official   503400.0
11  khangsresearchcentre1685   501700.0
12            albert_cancook   459100.0
13                   GoAmiGo   455300.0
14                    Vsauce   449000.0
15            gulshankalra07   431100.0
16                BayashiTV_   415900.0
17              fedevigevani   412200.0
18           IanLucasOficial   401700.0
19              mariano_razo   399300.0

esta tabela mostra os top 20 canais com mais like em todo mundo sendo a liderança de 5300000.0 like no primeiro canal de Mr Beast e em segundo canal também liderando o rank com 5000000.0  MrBeast2 sendo o youtuber mais influente em like da plataforma tendo somando seus dois canais terá o dobro que ultimo deste top 20 youtubers.

In [97]:
# Cria o grafico em barra do DF 'usuarios_por_like'
grafico_username = px.bar(usuarios_por_likes, x=usuarios_por_likes.Username, 
                          y=usuarios_por_likes.Likes, title='Top 20 Youtubers')
#Exibe o grafico
grafico_username.show()

### Tabela e gráfico relacionada com o número de visualizações.
Nessa partes nós fizemos o agrupamento do Dataframe para pegar a coluna 'Country' e somar as colunas 'Visits' e depois ordenar com as maiores visualizações por país.

In [98]:
# Agrupa o df pelo 'Country' e soma os valores da coluna 'Visitis' para cada pais
pais_por_visita = df.groupby('Country')['Visits'].sum().reset_index()

# Ordena o DF 'pais_por_visita' em decrescente com base na coluna 'Visitis'
pais_por_visita = pais_por_visita.sort_values(by='Visits', ascending=False)

# Redefine o indice 
pais_por_visita = pais_por_visita.reset_index(drop=True)

#Exibe o DF apos as operações
display(pais_por_visita)

Country       Visits
0   Estados Unidos  618549260.0
1            India  263282855.5
2          Unknown   76549900.0
3           México   61966100.0
4            Rusia   42040300.0
5           Brasil   26820400.0
6        Indonesia   20344100.0
7         Colombia   14566800.0
8        Argentina   14474200.0
9      El Salvador   12900000.0
10       Filipinas    9323200.0
11     Reino Unido    9250700.0
12            Iraq    6326700.0
13         Francia    6059100.0
14       Tailandia    4688600.0
15  Arabia Saudita    4574500.0
16           Japón    4195000.0
17         Turquía    3904900.0
18            Perú    2219400.0
19        Pakistán    2179200.0
20          España    2039900.0
21         Somalia    1900000.0
22         Argelia     524100.0
23          Egipto     326100.0
24        Jordania     267600.0
25      Bangladesh     100700.0
26         Ecuador      34300.0
27        Singapur      26400.0
28       Marruecos      12000.0

In [99]:
# Cria um grafico em barra usando o 'pais_por_visita'
grafico_visita_por_pais = px.bar(pais_por_visita, x=pais_por_visita.Country, 
                          y=pais_por_visita.Visits, title='Mais Visualizacoes por Pais')
# Exibe o grafico
grafico_visita_por_pais.show()

esta tabela representa número de visualizações de cada país sendo liderado por estados unidos em seguida a india com a diferença 3.552.664.045 bilhoes a mais que a índia.

### Tabela e Gráfico relacionada às quantidades de Canais por países

In [100]:
# Criar uma coluna 'IsAmerican' que indica se o canal é americano
df['IsAmerican'] = df['Country'] == 'Estados Unidos'

# Classificar o DataFrame com base na contagem de inscritos em ordem decrescente
df = df.sort_values(by='Subscribers_millions', ascending=False)

# Selecionar os top 1000 canais (ou os primeiros 1000 canais após a classificação)
top1000_canais = df.head(1000)

# Filtrar os canais americanos entre os top 1000
canais_americanos_top1000 = top1000_canais[top1000_canais['IsAmerican']]

# Contar quantos canais americanos estão no top 1000
quantidade_canais_americanos_top1000 = len(canais_americanos_top1000)

# Exibir a quantidade
print("Quantidade de canais americanos no top 1000:", quantidade_canais_americanos_top1000)

Quantidade de canais americanos no top 1000: 293


In [101]:
# Filtrar os canais americanos no top 1000
canais_americanos_top1000 = top1000_canais[top1000_canais['IsAmerican']]

# Contar a quantidade de canais americanos no top 1000
quantidade_canais_americanos_top1000 = len(canais_americanos_top1000)

# Filtrar os canais que não são americanos no top 1000
canais_outros_paises_top1000 = top1000_canais[top1000_canais['IsAmerican'] == False]

# Agrupar os canais por país (exceto os EUA) e contar a quantidade em cada país
contagem_canais_outros_paises = canais_outros_paises_top1000['Country'].value_counts().reset_index()
contagem_canais_outros_paises.columns = ['Country', 'Quantidade']

# Criar um DataFrame que inclua os EUA como um país separado com sua contagem
contagem_total_por_pais = pd.concat([pd.DataFrame({'Country': ['Estados Unidos'], 'Quantidade': [quantidade_canais_americanos_top1000]}), contagem_canais_outros_paises])

# Criar um gráfico de barras (bar chart) usando a contagem de canais por país
grafico = px.bar(contagem_total_por_pais, x='Country', y='Quantidade', 
                 labels={'Country': 'País', 'Quantidade': 'Quantidade de Canais'},
                 title='Quantidade de Canais por País no Top 1000')

# Exibir o gráfico de barras
grafico.show()

Dentre todos os canais do top1000 aproximadamente 300 são canais americanos, o que exemplifica mais uma vez a dominância do país. Por outro lado, mostra uma tendência de concentração, com apenas 3 países com mais de 100 canais.

## Conclusão

Sabemos que a ampla dominância dos EUA é em decorrência de seu alto publico, o que gera uma grande demanda em termos de publicidades, estamos falando de bilhões de visualizações, por isso muitas empresas estão dispostas a pagar para que anunciem seus produtos nesses canais.  Mas a plataforma pode investir para crescer fora dos Estados Unidos, avaliando mercados potenciais, como o brasileiro um mercado “emergente”, o indiano devido ao seu grande público. 
